### DISCUSSION TITLE
Srt delay solution

### DISCUSSION THREAD

**Discussion by esquilo444** [datarhei/restreamer#705] [Q&A] [answered]
Good morning,
Sorry for my many questions, but your project is so great and full of potential...
I see that the srt link at the bottom of the monitor is not the native srt that I send, it is a re-encoded version... Which necessarily generates a processing time... for my case around 1 second .
is it possible to retrieve the native srt stream that I send.
restreamer would be a simple relay server... which would greatly reduce the delay...
do you think this is feasible?
thanks in advance...
Jerome

**Comment by ioppermann** [comment]
Hi Jerome

Yes, you can use Restreamer as a SRT relay server. Just use any resource name in the streamid you want to publish a stream and use that same resource name when pulling the stream. There's no need to do anything in the UI.

In this example, I assume that you already enabled the SRT server and that you set a token `1234`. The resource name is `foobar`.

Publishing a stream: `srt://[ip:port]/?streamid=foobar,mode:publish,token:1234&mode=caller&transtype=live`
Pulling this stream: `srt://[ip:port]/?streamid=foobar,token:1234&mode=caller&transtype=live`

**Reply by esquilo444** [reply]
thank you very much for your quick response, I will test very soon...
if you need a beta tester, I will be happy to participate in the evolution of your project...

Sincerely

Jerome

**Reply by esquilo444** [reply]
Good morning,

I missed something... because the problem is still the same...
let me explain...
I send a stream encoded by a hardware box in HEVC using the SRT protocol... the restream server sees the stream and offers to re-encode it in HLS for monitoring the interface. So far so good...Then I recover the SRT stream that I want native, therefore without re-encoding, to VMIX in SRT with the tree essence that you communicated to me by selecting the srt stream in source. VMIX receives a SRT stream but it is still H264....So the reencoded version of my initial stream... Whereas I want to recover the HEVC stream via restreamer, but without reencoding... which is not the case... the goal would therefore be that restreamer for this use case behaves like a gateway, without any video processing... and thus drastically reduce the delay... Which according to my tests has not seems to be the case...
Thank you in advance for your light

Jerome

Ps maybe I missed an important adjustment... ;)

**Reply by ioppermann** [reply]
Let me better understand your setup. Do you use Restreamer to pull the stream via SRT from the hardware encoder, or does the hardware encoder send the stream via SRT to Restreamer? In the first case, the stream will get always encoded to H264. In the second case the SRT stream will not be changed.

**Reply by esquilo444** [reply]
this is the second case...i send my stream in srt from my encoder to restreamer...then i get it back from restreamer in srt to vmix...
the goal of my request is twofold, to save time and processing by not re-encoding the stream and also to have a source in caller mode and a final receiver in caller mode too... which greatly simplifies possible port forwarding problems on the transmitter's site as well as that of the site which receives the flow...
however, in my test I receive the stream on VMIX which tells me that it is h264, while the original stream is in h265. In addition the time of approximately 1 second between sending and receiving me makes it say that there is processing, because when I do point-to-point str I can easily go down to a delay of 150-250 ms.... I have been using srt in my work for several years and I have never had such a significant delay, except by choice by increasing my delay in the settings...

**Reply by ioppermann** [reply]
If vmix is also pulling the stream from Restreamer, then you actually don't need to do anything the UI. Like this Restreamer is acting plainly as a SRT server where you can send a stream to and one or several clients can pull the stream from.

Or do you use the SRT publication service to send the stream to vmix?

Referring to my first reply, in the encoder you use the streamid `foobar,mode:publish,token:1234`, whereas `foobar` is the name of the stream (you give it any name you want). The token is something you choose when you enable the SRT server. In vmix you use the streamid `foobar,token:1234` in order to pull in the stream.

After your encoder sends the stream to Restreamer, don't do anything in the UI.

**Reply by esquilo444** [reply]
I do exactly as you say.
There are just two things I don't understand:
- the SRT delay between transmitter and final receiver via restreamer is 4 times greater than a point-to-point SRT.
- vmix tells me that it is an h264 that it receives even though the source is in h265... I use the architecture that you gave me...
I must have made a mistake somewhere...I can't identify it.

**Reply by esquilo444** [reply]
Good morning,

I continue to work on my problem. Which is progressing quite well... with your instructions I actually managed to recover the native h265 stream via restreamer on a receiving PC, it works very well on OBS, but for the moment, impossible to receive on the same PC via VMIX.. I haven't found why yet...
In any case, thank you again for taking the time with me...

Jerome

**Reply by ioppermann** [reply]
Maybe vmix is not able to detect the stream codec properly.

**Reply by esquilo444** [reply]
I don't think so because, when I don't use restream, so from point to point, it works perfectly with the same encoding characteristics

**Reply by esquilo444** [reply]
Good,
After other tests..following an error on my part, in my testing process. the problem really comes from VMIX... so I'm going to investigate on this side... I'll let you know as soon as I find a solution...

**Reply by esquilo444** [reply, marked as answer]
Good,
Solution found...
VMIX does not use its own driver to decode h265 unlike OBS... I had to install the proprietary HEVC codec of Windows 11... And then miracle.....
Thanks again for the time spent...
I remain at your disposal, if you have any questions or if you are looking for beta testers...

Good day.

Jerome

**Reply by ioppermann** [reply]
Good the hear that it worked out.

**Reply by esquilo444** [reply]
thank you...
You always have a solution...
I would like to take this opportunity to ask you if you have a list of compatible acquisition cards to input a local signal to a restreamer...
I know I always ask lots of questions, but I have so many ideas for using your server that I always want more 😅
